christfellowshipchurch/web-app
# Group Chat

# Related Issue
- [**🏕 Group Chat**](https://3.basecamp.com/3926363/buckets/17566548/documents/3094822151)

# Changes or Updates
- Duplicates `GroupContentItem` folder as `NewGroupContentItem`, imported when `sandbox` mode is enabled
- Adds support for Group Chat
- Cleans up some import paths and naming in groups components

# Test instructions
1. Run your local **🛰️ api** on the `group-chat` branch
1. Ensure `REACT_APP_ENABLE_SANDBOX=true` in your local web app `.env`
1. Start the web server
1. Log in as a user with access to the `GROUP_CHAT` feature flag
1. Navigate to http://local.christfellowship.church:3000/groups
1. 🔸 Verify the groups list page renders properly
1. Click a group's card
1. 🔸 Verify the group details page uses the new WIP layout, with Group Chat enabled
1. Stop the web server
1. Ensure `REACT_APP_ENABLE_SANDBOX=` in your local web app `.env`
1. Repeat steps above, but 🔸 Verify the old/existing groups page is rendered instead

# Screenshots
|Production/Standard Version|Sandbox Version|
|---|---|
|![CleanShot 2020-10-20 at 13 26 10@2x](https://user-images.githubusercontent.com/1812955/96622631-32d52f00-12d8-11eb-92a1-00c377d28741.jpg)|![CleanShot 2020-10-20 at 13 24 00@2x](https://user-images.githubusercontent.com/1812955/96622618-2ea91180-12d8-11eb-8287-bcd8b4c554bf.jpg)|

# Tests
